shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -159,32 +206,56 @@ class FileChooserDialog {
   CHROMEG_CALLBACK_1(FileChooserDialog,
                      void,
                      OnFileDialogResponse,
-                     GtkWidget*,
+                     GtkFileChooser*,
                      int);
 
-  GtkWidget* dialog() const { return dialog_; }
+  GtkFileChooser* dialog() const { return dialog_; }
+  GModule* gtk_module() const { return gtk_module_; }
 
  private:
   void AddFilters(const Filters& filters);
 
   electron::NativeWindowViews* parent_;
   electron::UnresponsiveSuppressor unresponsive_suppressor_;
 
-  GtkWidget* dialog_;
+  GtkFileChooser* dialog_;
   GtkWidget* preview_;
+  GModule* gtk_module_;
 
   Filters filters_;
   std::unique_ptr<electron::util::Promise> save_promise_;
   std::unique_ptr<electron::util::Promise> open_promise_;
 
   // Callback for when we update the preview for the selection.
-  CHROMEG_CALLBACK_0(FileChooserDialog, void, OnUpdatePreview, GtkWidget*);
+  CHROMEG_CALLBACK_0(FileChooserDialog, void, OnUpdatePreview, GtkFileChooser*);
+
+  // Sets |dialog| as transient for |parent|, which will keep it on top and
+  // center it above |parent|. Do nothing if |parent| is nullptr.
+  void SetGtkTransientForAura(aura::Window* parent) {
+    if (!parent || !parent->GetHost())
+      return;
```

Comment:
I don't see the `!parent->GetHost()` call in the earlier version. Has something changed to cause this test to be needed?

Also, if we do need it, should we warn on failure with `G_RETURN_IF_FAIL(!parent->GetHost() != nullptr);` ?

Reply:
I am referring to the function located in the Chromium code base: https://github.com/chromium/chromium/blob/master/chrome/browser/ui/libgtkui/gtk_util.cc#L136. Is this wrong?

Reply:
OK, I was comparing it to the previous version of `shell/browser/ui/file_dialog_gtk.cc`. I don't mind the extra check being there but wanted to know _why_ it was there. Adopting from libgtkui WfM.